src/es6/Entity.js

```diff
@@ -82,10 +89,24 @@ export class Entity {
 		if (this._subEntities.has(source)) {
 			dispose(this._subEntities.get(source));
 		}
-		entityFactory(entityType, source, this._token, (entity) => {
-			this._subEntities.set(source, entity);
-			onChange(entity);
-		});
+
+		this._subEntitiesLoadStatus.push(new Promise((resolve, reject) => {
+			entityFactory(entityType, source, this._token, (entity, error) => {
+				this._subEntities.set(source, entity);
+				onChange(entity, error);
+				if (error) {
+					reject && reject();
+					resolve = null;
+					reject = null;
+				} else {
+					Promise.all(entity._subEntitiesLoadStatus).then(() => {
+						resolve && resolve();
+						resolve = null;
+						reject = null;
```

Comment:
Setting these to null because it has been rumoured to cause problems if you do not. I am assuming polyfills get angry. So better be safe.